pkg/k8shandler/elasticsearch_helper.go

```diff
@@ -0,0 +1,369 @@
+package k8shandler
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	certLocalPath = "/tmp/"
+)
+
+type esCurlStruct struct {
+	Method       string // use net/http constants https://golang.org/pkg/net/http/#pkg-constants
+	URI          string
+	RequestBody  string
+	StatusCode   int
+	ResponseBody map[string]interface{}
+	Error        error
+}
+
+func parseBool(path string, interfaceMap map[string]interface{}) bool {
+	value := walkInterfaceMap(path, interfaceMap)
+
+	if parsedBool, ok := value.(bool); ok {
+		return parsedBool
+	} else {
+		return false
+	}
+}
+
+func parseString(path string, interfaceMap map[string]interface{}) string {
+	value := walkInterfaceMap(path, interfaceMap)
+
+	if parsedString, ok := value.(string); ok {
+		return parsedString
+	} else {
+		return ""
+	}
+}
+
+func parseInt32(path string, interfaceMap map[string]interface{}) int32 {
+	return int32(parseFloat64(path, interfaceMap))
+}
+
+func parseFloat64(path string, interfaceMap map[string]interface{}) float64 {
+	value := walkInterfaceMap(path, interfaceMap)
+
+	if parsedFloat, ok := value.(float64); ok {
+		return parsedFloat
+	} else {
+		return float64(-1)
+	}
+}
+
+func walkInterfaceMap(path string, interfaceMap map[string]interface{}) interface{} {
+
+	current := interfaceMap
+	keys := strings.Split(path, ".")
+	keyCount := len(keys)
+
+	for index, key := range keys {
+		if current[key] != nil {
+			if index+1 < keyCount {
+				current = current[key].(map[string]interface{})
+			} else {
+				return current[key]
+			}
+		} else {
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func updateAllIndexReplicas(clusterName, namespace string, client client.Client, replicaCount int32) (bool, error) {
```

Comment:
A discussion that came up with SRE as an escape hatch for consideration. How do we allow customization here for cases which warrant variation?  Do we need to consider whitelisting?  Do customers need to go Unmanaged?

Reply:
What sort of customization would we be doing? `Unmanaged` is the escape hatch for anything custom - the operator wouldn't adjust the redundancy on the cluster if unmanaged.